src/lib/client/GoodCityApiV2Client/GoodCityApiV2Client.test.ts

```diff
@@ -0,0 +1,79 @@
+import GoodCityApiV2Client from "./GoodCityApiV2Client";
+import sendPinResponse from "./__mocks__/sendPinResponse.js";
+import verifyResponse from "./__mocks__/verifyResponse.js";
+
+const mockFetch = jest.spyOn(window, "fetch");
+
+afterEach(() => mockFetch.mockClear());
+
+afterAll(() => mockFetch.mockRestore());
+
+describe("sendPin", () => {
```

Comment:
Any ideas on a better structure for the tests? The current structure leads to a lot of duplication when new endpoints are added.
First of all we want to assert each method calls fetch correctly.
Secondly, we want to assert on the different type of responses which include
1. Successful API response. (we need to assert on the data format)
2. Error response from server e.g. (422 Mobile number invalid)
3. Network Error (e.g. request failed to reach server)

Among the above 1 and 2 are endpoint specific whereas Network Error is generic.
For 1 and 2 we need to create mocks and type definitions for the success case and error case. Any way to make this easier (e.g. autogeneration)?